proposals/relaxed-simd/entropy.md

```diff
@@ -0,0 +1,57 @@
+# Entropy, Compat and Usage patterns
+
+The Relaxed SIMD proposal adds new operations that take advantage of the underlying hardware for accelerated performance by loosening the portability constraints of fixed-width SIMD, or adding new instructions that introduce local non-determinisim. This introduces a few potential risks specifically for browser engine implementations, though it’s possible that one or more of these might be applicable to other environments: 
+* Identifying underlying characteristics of the device (processor information if the engine that implements this proposal used the most optimal lowerings)
+* Possibly identifying information about the browser currently in use (i.e. if a sufficiently motivated user writes a hand tuned Wasm function when engines use different lowerings for instructions)
+* Compatibility risks, i.e. if code compiled for one browser works differently on a different browser (This is specific to observable behavior changes, and not performance differences)
+
+This document is an attempt to collate information that is already available in the issues filed for each of the operations on fingerprinting risk, quantify the risk for fingerprinting/compat issues and provide some information about usage patterns. 
+
+## Instruction summary
+
+* **relaxed i8x16.swizzle**<br>
+  *Entropy exposed:* Differences between Intel/ARM/RISC-V<br>
+  *Deterministic lowering:* Available<br>
+  *Compat risk:* Low, as the differences exposed are for out of range indices<br>
+  [*Issue link*](https://github.com/WebAssembly/relaxed-simd/issues/21)
+* **relaxed i32x4.trunc_{f32x4, f64x2} operations**<br>
+  *Entropy exposed:* Differences between Intel/ARM/RISC-V<br>
+  *Deterministic lowering:* Available<br>
+  *Compat risk:* Low, as the differing behavior is for out of range values, and NaNs<br>
+  [*Issue link*](https://github.com/WebAssembly/relaxed-simd/issues/21)<br>
+* **qfma, qfmns**<br>
+  *Entropy exposed:* Differences between hardware that has native FMA support, and hardware that does not.<br>
+  *Deterministic lowering:* Not available, but depending on underlying hardware, the results can only be fused, or unfused.<br>
+  *Compat risk:* TBD<br>
+  *Mitigating reasons to include:* Most modern hardware do have native FMA support, performance wins are significant, operations are difficult to emulate<br>
+  [*Issue link*](https://github.com/WebAssembly/simd/pull/79)<br>
+* **{i8x16, i16x8, i32x4, i64x2}.laneselect**<br>
+  *Entropy exposed:* Intel/ARM<br>
+  *Deterministic lowering:* Available<br>
+  *Compat risk:* Medium, architectures vary on which bit is used for lane selection<br>
+  [*Issue link*](https://github.com/WebAssembly/relaxed-simd/issues/17)<br>
+* **{f32x4, f64x2}.{min,max}**<br>
+  *Entropy exposed:* Intel/ARM<br>
+  *Deterministic lowering:* Available<br>
+  *Compat risk:* Low, varying outputs when one of the inputs is NaN, or  +0, -0<br>
```

Comment:
In case of NaNs, the difference can be NaN vs a value, I think that is somewhat substantial.

Reply:
By difference do you mean the output? Substantial in what sense?

Reply:
On x86 fmin&fmax return the other value for single NaN input, while Arm would return a NaN (IEEE-compatible semantics), this way the same code can end up with NaN on one platform and with a value on the other. From point of view of getting a valid number as an answer this is significant difference.